src/js/views/maps/LayerDetailsView.js

```diff
@@ -206,6 +215,15 @@ define([
               detailSection.el.style.display = "none";
             }
           }
+          // Hide the Filter by Property view if there is no filter model for the selected layer
```

Comment:
Here we have to depend on the label of the section: "Filter by Property". If we change that label in the future, then we have to remember to change it here too. It makes the code fragile. To make it more robust, we could use the `renderIf` function mentioned above. Right before we create the `detailSection` we can return if the `renderIf` function returns false. e.g.:

```js
// If the section has a renderIf function, check if we should render
// this section or not
if (
  typeof section.renderIf === "function" &&
  !section.renderIf(model)
) {
  return null; // Skip rendering this section
}
```